easybuild/easyconfigs/q/Qt/Qt-4.8.7_openssl_1.1.patch

```diff
@@ -0,0 +1,694 @@
++#endif
+     RESOLVEFUNC(DSA_free, 594, libs.second )
+     RESOLVEFUNC(ERR_error_string, 744, libs.second )
+     RESOLVEFUNC(ERR_get_error, 749, libs.second )
+@@ -678,8 +738,10 @@
+     RESOLVEFUNC(SSL_get_peer_cert_chain, 117, libs.first )
+     RESOLVEFUNC(SSL_get_peer_certificate, 118, libs.first )
+     RESOLVEFUNC(SSL_get_verify_result, 132, libs.first )
++#if OPENSSL_VERSION_NUMBER < 0x10100000L
+     RESOLVEFUNC(SSL_library_init, 137, libs.first )
+     RESOLVEFUNC(SSL_load_error_strings, 139, libs.first )
++#endif
+     RESOLVEFUNC(SSL_new, 140, libs.first )
+ #if OPENSSL_VERSION_NUMBER >= 0x0090806fL && !defined(OPENSSL_NO_TLSEXT)
+     RESOLVEFUNC(SSL_ctrl, 95, libs.first )
+@@ -751,9 +813,11 @@
+     RESOLVEFUNC(BIO_write)
+     RESOLVEFUNC(BN_num_bits)
+     RESOLVEFUNC(CRYPTO_free)
++#if OPENSSL_VERSION_NUMBER < 0x10100000L
+     RESOLVEFUNC(CRYPTO_num_locks)
+     RESOLVEFUNC(CRYPTO_set_id_callback)
+     RESOLVEFUNC(CRYPTO_set_locking_callback)
++#endif
+     RESOLVEFUNC(DSA_free)
+     RESOLVEFUNC(ERR_error_string)
+     RESOLVEFUNC(ERR_get_error)
+@@ -783,10 +847,17 @@
+     RESOLVEFUNC(RAND_seed)
+     RESOLVEFUNC(RAND_status)
+     RESOLVEFUNC(RSA_free)
++#if OPENSSL_VERSION_NUMBER < 0x10100000L
+     RESOLVEFUNC(sk_free)
+     RESOLVEFUNC(sk_num)
+     RESOLVEFUNC(sk_pop_free)
+     RESOLVEFUNC(sk_value)
++#else
++    RESOLVEFUNC(OPENSSL_sk_free)
++    RESOLVEFUNC(OPENSSL_sk_num)
++    RESOLVEFUNC(OPENSSL_sk_pop_free)
++    RESOLVEFUNC(OPENSSL_sk_value)
++#endif
+     RESOLVEFUNC(SSL_CIPHER_description)
+     RESOLVEFUNC(SSL_CTX_check_private_key)
+     RESOLVEFUNC(SSL_CTX_ctrl)
+@@ -801,6 +872,7 @@
+     RESOLVEFUNC(SSL_CTX_use_PrivateKey)
+     RESOLVEFUNC(SSL_CTX_use_RSAPrivateKey)
+     RESOLVEFUNC(SSL_CTX_use_PrivateKey_file)
++    RESOLVEFUNC(SSL_CTX_get_cert_store)
+     RESOLVEFUNC(SSL_accept)
+     RESOLVEFUNC(SSL_clear)
+     RESOLVEFUNC(SSL_connect)
+@@ -811,8 +883,12 @@
+     RESOLVEFUNC(SSL_get_peer_cert_chain)
+     RESOLVEFUNC(SSL_get_peer_certificate)
+     RESOLVEFUNC(SSL_get_verify_result)
++#if OPENSSL_VERSION_NUMBER < 0x10100000L
+     RESOLVEFUNC(SSL_library_init)
+     RESOLVEFUNC(SSL_load_error_strings)
++#else
++    RESOLVEFUNC(OPENSSL_init_ssl)
++#endif
+     RESOLVEFUNC(SSL_new)
+ #if OPENSSL_VERSION_NUMBER >= 0x0090806fL && !defined(OPENSSL_NO_TLSEXT)
+     RESOLVEFUNC(SSL_ctrl)
+@@ -823,13 +899,39 @@
+     RESOLVEFUNC(SSL_set_connect_state)
+     RESOLVEFUNC(SSL_shutdown)
+     RESOLVEFUNC(SSL_write)
++
++    RESOLVEFUNC(X509_get_serialNumber)
++#if OPENSSL_VERSION_NUMBER >= 0x10100000L
++    RESOLVEFUNC(SSL_CTX_ctrl)
++    RESOLVEFUNC(EVP_PKEY_id)
++    RESOLVEFUNC(EVP_PKEY_base_id)
++    RESOLVEFUNC(SSL_CIPHER_get_bits)
++    RESOLVEFUNC(SSL_CTX_set_options)
++    RESOLVEFUNC(X509_get_version)
++    RESOLVEFUNC(X509_get_X509_PUBKEY)
++    RESOLVEFUNC(RSA_bits)
++    RESOLVEFUNC(DSA_security_bits)
++    RESOLVEFUNC(DSA_get0_pqg)
++#if OPENSSL_VERSION_NUMBER < 0x10100000L
```

Comment:
Hehe, nothing I touched, either.. You are right, this would never be executed. The patch has been included in Debian's Qt4 build, as noted, but apparently it was never properly reviewed. If you think the quality is not hight enought to include it into easybuild, that's completely fine with me. Apparently there are not too many complaints for Qt4 not being readily available.

One of the `#endif``s below the block could be moved up, but I have no idea how to properly verify, that everything is working as expected, then.